tutorials/source-modeling/plot_eeg_mri_coords.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+"""
+.. _tut-eeg-mri-coords:
+
+EEG source localization given electrode locations on an MRI
+===========================================================
+
+This tutorial explains how to compute the forward operator from EEG data
+when the electrodes are in MRI voxel coordinates.
+
+.. contents:: This tutorial covers:
+   :local:
+   :depth: 2
+
+"""
+
+# Authors: Eric Larson <larson.eric.d@gmail.com>
+#
+# License: BSD Style.
+
+import os.path as op
+
+import nibabel
+from nilearn.plotting import plot_glass_brain
+import numpy as np
+
+import mne
+from mne.channels import make_dig_montage, compute_native_head_t
+from mne.io.constants import FIFF
+from mne.transforms import apply_trans
+from mne.viz import plot_alignment
+
+data_path = mne.datasets.sample.data_path()
+subjects_dir = op.join(data_path, 'subjects')
+fname_T1 = op.join(subjects_dir, 'sample', 'mri', 'T1.mgz')
+
+##############################################################################
+# Faking an appropriate EEG-MRI dataset
+# -------------------------------------
+# Let's first manipulate ``sample`` to give us:
+#
+# - ``eeg_vox``: EEG electrodes in RAS voxel coordinates
+# - ``fid_vox``: fiducial (LPA, nasion, RPA) locations in RAS voxel coordinates
+# - ``raw``: a Raw instance with only EEG channels.
+#
+# This is just so we have data to work with -- hopefully with real EEG+MRI
+# datasets, these values are obtainable.
+
+trans = mne.read_trans(
+    op.join(data_path, 'MEG', 'sample', 'sample_audvis_raw-trans.fif'))
+raw = mne.io.read_raw_fif(
+    op.join(data_path, 'MEG', 'sample', 'sample_audvis_raw.fif'))
+# Get fiducials and EEG locations in MRI voxel coords
+trans_mm = trans['trans'].copy()
+trans_mm[:3, 3] *= 1e3
+img = nibabel.load(fname_T1)
+vox2mri = img.header.get_vox2ras_tkr()
+head_to_vox = np.dot(np.linalg.inv(vox2mri), trans_mm)
+fid_vox = apply_trans(head_to_vox, 1000 * np.array([d['r'] for d in sorted(
+    [d for d in raw.info['dig'] if d['kind'] == FIFF.FIFFV_POINT_CARDINAL],
+    key=lambda d: d['ident'])]))
+eeg_vox = apply_trans(head_to_vox, 1000 * np.array(
+    [d['r'] for d in raw.info['dig'] if d['kind'] == FIFF.FIFFV_POINT_EEG and
+     d['ident'] > 0]))
+img, affine = img.get_data().astype(float), img.affine
+# Mark them in the MRI with 5x5x5mm cubes
+for coord in np.round(np.concatenate([fid_vox, eeg_vox])).astype(int):
+    img[tuple(slice(c - 2, c + 2) for c in coord)] = 250
+# change "raw" to be more like what you'd acquire: no dig, no channel locations
+raw.load_data().pick_types(meg=False, eeg=True, exclude=(), stim=True)
+raw.info['dig'] = []
+for ch in raw.info['chs']:
+    ch['loc'].fill(0.)
+del head_to_vox, trans
```

Comment:
I am not super happy about exposing such complex / low level code. Can we have an alternative?